astropy/table/tests/test_index.py

```diff
@@ -460,3 +477,34 @@ def test_copy_indexed_table(self, table_types):
             for index, indexp in zip(t.indices, tp.indices):
                 assert np.all(index.data.data == indexp.data.data)
                 assert index.data.data.colnames == indexp.data.data.colnames
+
+    def test_updating_row_byindex(self, main_col, table_types, engine):
+        self._setup(main_col, table_types)
+        t = Table([[1, 2, 3, 4], [2, 3, 4, 5], [3, 4, 5, 6]], names=('a', 'b', 'c'), meta={'name': 'first table'})
+
+        t.add_index('a', engine=engine)
+        t.add_index('b', engine=engine)
+
+        t.loc[3] = [3,4,5] # single label, with primary key 'a'
+        t2 = t[2]
+        assert list(t2) == [3, 4, 5]
+
+        # list search
+        t.loc[[1, 4, 2]] = [[1, 2, 3], [4, 5, 6], [2, 3, 4]]
```

Comment:
Same comment, it seems you are setting to the original values so the test is not good.